src/registry/config.rs

```diff
@@ -50,6 +62,45 @@ impl DockerConfig {
 
         Ok(self.auths.get(reference.registry()).cloned())
     }
+
+    pub fn get_auth_from_credentials_helper_if_present(
```

Comment:
nit, adding some documentation about what the method does would be nice.

Something has short as:

```suggestion
    /// Get registry credentials using an external credential-helper.
    /// Returns `None` when no credential-helper is configured for the requested registry
    /// Otherwise returns `Some`-thing when a credential-helper is configured. The inner object
    /// can be an `Err` if something went wrong while invoking the external helper
    pub fn get_auth_from_credentials_helper_if_present(
```